datadog_checks_base/tests/test_http.py

```diff
@@ -80,12 +87,19 @@ def test_config_init_config_override(self):
 
 
 class TestHeaders:
+    def test_agent_headers(self):
```

Comment:
We didn't have a unit test for this before, yet this is something that the `http_check` uses:

https://github.com/DataDog/integrations-core/blob/c1f7fd7220a25ed0c1e10aa8306fcd7267a26b3d/http_check/datadog_checks/http_check/config.py#L49-L50

Maybe it should use something else (like `get_default_headers()`?), but in the meantime the `http_check` will automatically benefit from these fixes w/o having to change anything there.